cohortextractor/main.py

```diff
@@ -1,18 +1,33 @@
 import csv
 import importlib.util
 import inspect
+import shutil
 import sys
 from contextlib import contextmanager
+from pathlib import Path
 
-from .backends import BACKENDS
-from .query_utils import get_column_definitions
+from cohortextractor.backends import BACKENDS
+from cohortextractor.query_utils import get_column_definitions
+from cohortextractor.validate_dummy_data import (
+    DummyDataValidationError,
+    validate_dummy_data,
+)
 
 
-def main(definition_path, output_file, backend_id, db_url):
-    backend = BACKENDS[backend_id](db_url)
+def main(definition_path, output_file, backend_id, db_url, dummy_data_file=None):
     cohort = load_cohort(definition_path)
-    results = extract(cohort, backend)
-    write_output(results, output_file)
+
+    if dummy_data_file:
+        results = None
+        if Path(output_file).suffixes != dummy_data_file.suffixes:
+            expected_extension = "".join(output_file.suffixes)
+            msg = f"Expected dummy data file with extension {expected_extension}; got {dummy_data_file}"
+            raise DummyDataValidationError(msg)
+        validate_dummy_data(get_column_definitions(cohort), dummy_data_file)
```

Comment:
This is a tiny thing, but I would move the call to `get_column_definitions()` into the validation function. Again to keep the stuff here at the same level of abstraction.